File: src/main/java/com/library/users/Etudiant.java
package com.library.users;

/**
 * Classe Étudiant
 * Représente un étudiant qui peut emprunter jusqu'à 3 livres
 */
public class Etudiant extends Utilisateur {
    private String numeroEtudiant;
    private static final int MAX_EMPRUNTS = 3;

    public Etudiant(String id, String nom, String email, String numeroEtudiant) {
        super(id, nom, email);
        this.numeroEtudiant = numeroEtudiant;
    }

    public String getNumeroEtudiant() {
        return numeroEtudiant;
    }

    @Override
    protected int getMaxEmprunt() {
        return MAX_EMPRUNTS;
    }

    @Override
    public String toString() {
        return "Étudiant{" +
                "id='" + getId() + '\'' +
                ", nom='" + getNom() + '\'' +
                ", email='" + getEmail() + '\'' +
                ", numeroEtudiant='" + numeroEtudiant + '\'' +
                ", livresEmpruntés=" + getNombreLivresEmpruntés() + "/" + MAX_EMPRUNTS +
                '}';
    }
}

File: js/src/users/Etudiant.js
const Utilisateur = require('./Utilisateur');

/**
 * Classe Étudiant
 * Représente un étudiant qui peut emprunter jusqu'à 3 livres
 */
class Etudiant extends Utilisateur {
    constructor(id, nom, email, numeroEtudiant) {
        super(id, nom, email);
        this.numeroEtudiant = numeroEtudiant;
    }

    static MAX_EMPRUNTS = 3;

    getMaxEmprunt() {
        return Etudiant.MAX_EMPRUNTS;
    }

    getNumeroEtudiant() {
        return this.numeroEtudiant;
    }

    toString() {
        return `Étudiant{
            id: '${this.id}',
            nom: '${this.nom}',
            email: '${this.email}',
            numeroEtudiant: '${this.numeroEtudiant}',
            livresEmpruntés: ${this.getNombreLivresEmpruntés()}/${Etudiant.MAX_EMPRUNTS}
        }`;
    }
}

module.exports = Etudiant;
